File: backend-php/utils/PasswordUtil.php
<?php
/**
 * Hachage et vérification des mots de passe avec BCrypt (cost=12).
 * Équivalent PHP de com.ihecride.utils.PasswordUtil.
 */

class PasswordUtil {

    const COST = 12;

    public static function hash(string $clearPassword): string {
        return password_hash($clearPassword, PASSWORD_BCRYPT, ['cost' => self::COST]);
    }

    public static function verify(?string $clearPassword, ?string $hash): bool {
        if ($clearPassword === null || $hash === null) return false;
        try {
            return password_verify($clearPassword, $hash);
        } catch (\Throwable $e) {
            return false;
        }
    }

    /**
     * Robustesse : min. 8 caractères, 1 majuscule, 1 chiffre, 1 caractère spécial.
     */
    public static function isStrong(?string $pwd): bool {
        if ($pwd === null || strlen($pwd) < 8) return false;
        return preg_match('/[A-Z]/', $pwd)
            && preg_match('/\d/', $pwd)
            && preg_match('/[!@#$%^&*(),.?":{}|<>]/', $pwd);
    }
}

File: backend/src/main/java/com/ihecride/utils/PasswordUtil.java
package com.ihecride.utils;

import org.mindrot.jbcrypt.BCrypt;

/**
 * Hachage et vérification des mots de passe avec BCrypt (facteur 12).
 */
public class PasswordUtil {

    private static final int SALT_ROUNDS = 12;

    public static String hash(String clearPassword) {
        return BCrypt.hashpw(clearPassword, BCrypt.gensalt(SALT_ROUNDS));
    }

    public static boolean verify(String clearPassword, String hash) {
        if (clearPassword == null || hash == null) return false;
        try {
            return BCrypt.checkpw(clearPassword, hash);
        } catch (Exception e) {
            return false;
        }
    }

    /**
     * Vérifie la robustesse :
     * min. 8 caractères, 1 majuscule, 1 chiffre, 1 caractère spécial.
     */
    public static boolean isStrong(String pwd) {
        if (pwd == null || pwd.length() < 8) return false;
        return pwd.matches(".*[A-Z].*") &&
               pwd.matches(".*\\d.*") &&
               pwd.matches(".*[!@#$%^&*(),.?\":{}|<>].*");
    }
}
